### PR TITLE
fix: debug and fix lost messages

### PR DESCRIPTION
**Pull request by aeddi**

### CODE REVIEW

File: pkg/cryptoutil/keystore_message.go [hunk 1]
```diff
@@ -791,12 +798,12 @@ func (m *MessageKeystore) UpdatePushGroupReferences(ctx context.Context, deviceP
...
 		ref, err := CreatePushGroupReference(devicePK, refsToCreate[i], groupPushSecret)
 		if err != nil {
-			// TODO: log
+			logger.Error("creating new push group reference failed", logutil.PrivateBinary("ref", ref), zap.Error(err))
 			continue
 		}
 
 		if err := m.store.Put(ctx, m.refKey(ref), group.GetPublicKey()); err != nil {
-			// TODO: log
+			logger.Error("deleting new push group reference failed", logutil.PrivateBinary("ref", ref), zap.Error(err))
```
Comment (D4ryl00) [inline pkg/cryptoutil/keystore_message.go:806]: Is it `deleting` or `storing`?

File: pkg/cryptoutil/keystore_message.go [hunk 2]
```diff
@@ -330,47 +335,49 @@ type computedKey struct {
 }
 
 func (m *MessageKeystore) putPrecomputedKeys(ctx context.Context, groupPK, device crypto.PubKey, preComputedKeys ...computedKey) error {
-	if m == nil || len(preComputedKeys) == 0 {
+	if m == nil {
```
Comment (D4ryl00) [inline pkg/cryptoutil/keystore_message.go:338]: So `len(preComputedKeys) == 0` doesn't return an error any more, is it what you expect for?

File: service_group.go [hunk 3]
```diff
@@ -142,7 +142,7 @@ func (s *service) activateGroup(ctx context.Context, pk crypto.PubKey, localOnly
 	dbOpts := &iface.CreateDBOptions{LocalOnly: &localOnly}
 	gc, err := s.odb.OpenGroup(ctx, g, dbOpts)
 	if err != nil {
-		return errcode.ErrGroupOpen.Wrap(err)
+		return err
```
Comment (D4ryl00) [inline service_group.go:145]: why did you remove this errcode? Should you also remove this errcode in the errcode.proto file?

**Review by D4ryl00 — APPROVED** [review]

File: pkg/cryptoutil/keystore_message.go (hunk 2 above)
Comment (aeddi-sudo) [reply pkg/cryptoutil/keystore_message.go:338]: Yes that's the point of this fix

Comment (aeddi) [reply pkg/cryptoutil/keystore_message.go:338]: Yes that's the point of the fix

File: pkg/cryptoutil/keystore_message.go (hunk 1 above)
Comment (aeddi) [reply pkg/cryptoutil/keystore_message.go:806]: Yes copy/paster error

File: pkg/cryptoutil/keystore_message.go [hunk 4]
```diff
@@ -644,18 +652,19 @@ func (m *MessageKeystore) updateCurrentKey(ctx context.Context, groupPK, pk cryp
 }
 
 // NewMessageKeystore instantiate a new MessageKeystore
-func NewMessageKeystore(s datastore.Datastore) *MessageKeystore {
+func NewMessageKeystore(s datastore.Datastore, logger *zap.Logger) *MessageKeystore {
 	return &MessageKeystore{
 		preComputedKeysCount: 100,
 		store:                dssync.MutexWrap(s),
+		logger:               logger.Named("message-ks"),
```
Comment (D4ryl00) [inline pkg/cryptoutil/keystore_message.go:659]: Shouldn't you test if the logger is nil? If it is, instantiate a new one (or `NewNop()`), else set a new namespace (`logger = logger.Named("keystore")`).
If you don't do that, you will get a runtime error when you'll want to log.

File: pkg/cryptoutil/keystore_message.go [hunk 5]
```diff
@@ -644,18 +652,19 @@ func (m *MessageKeystore) updateCurrentKey(ctx context.Context, groupPK, pk cryp
...
 	}
 }
 
 // nolint:deadcode,unused // NewInMemMessageKeystore instantiate a new MessageKeystore, useful for testing
-func NewInMemMessageKeystore() (*MessageKeystore, func()) {
+func NewInMemMessageKeystore(logger *zap.Logger) (*MessageKeystore, func()) {
 	ds := dssync.MutexWrap(datastore.NewMapDatastore())
 
-	return NewMessageKeystore(ds), func() { _ = ds.Close() }
+	return NewMessageKeystore(ds, logger), func() { _ = ds.Close() }
```
Comment (D4ryl00) [inline pkg/cryptoutil/keystore_message.go:667]: Same as `NewMessageKeystore`.

**Review by D4ryl00 — APPROVED** [review]

File: pkg/cryptoutil/keystore_message.go (hunk 4 above)
Comment (aeddi) [reply pkg/cryptoutil/keystore_message.go:663]: Yep
